### ISSUE TITLE
`Application::t()` is too long

### ISSUE THREAD

**Issue by gagatust** [yiisoft/core#4]
See https://github.com/yiisoft/core/commit/c6e0e031d419881d1c02d4894155e94516da1a26
`Application::t()` is too long than `Yii::t()`.This method is often used. What about `App::t()`?

**Comment by hiqsol** [comment]
Please note it's not static.
To be used: `$this->app->t(...)`
And `Yii::t()` is not removed.

**Comment by gagatust** [comment]
What is better `$this->app->t()` or `Yii::t()` and why?

**Comment by SilverFire** [comment]
The main point is "destatification" inside the framework core. While static methods can serve as good shortcuts for faster code composing, they encourage developers to rely on global application state, that is usually not a good thing. You don't like to use `$_GLOBALS` variable, right?

There are some posts [[1]](https://www.yegor256.com/2014/05/05/oop-alternative-to-utility-classes.html), [[2]](https://r.je/static-methods-bad-practice.html) that explain problem a bit wider.

**Comment by gagatust** [comment]
Do you plan to remove `Yii::t()`?

**Comment by hiqsol** [comment]
Not now. And only after having good alternative.

**Comment by hiqsol** [comment]
At the moment we are only trying to clean up the framework core.

**Comment by tecnoterabyte** [comment]
Excelent  "destatification" helpers also.

**Comment by slinstj** [comment]
Hi guys!

About the "destatification", as I was reading the pointed articles, I tended to discord of them. I think it could be just "over-engineering" to simple situations. When I got to the end of the article, many comments came with strong variated contrary arguments ([like this](https://www.yegor256.com/2014/05/05/oop-alternative-to-utility-classes.html#comment-3476896668)).

I think the "**Y**eah, **i**t **i**s" could be lost if Yii3 "dificult" some simple things. And helpers are this: simple things. They have no state. They dont change states. They are simple functions.

It would be great to hear really "strong" arguments to remove (or substitute) static helpers, because I read the pointed articles and I could not find them (nor the most of the commenters). Hope to hear you here... thank you.

**Comment by gagatust** [comment]
@slinstj I think the problem is that Yii::t() is not pure function. Yii:t() behavior depends on settings. We can change the settings at any time and change the behavior.

**Comment by hiqsol** [comment]
@slinstj 
First of all `Yii::t()` is not about to be removed.
But things are being cleaned up.

`$this->app->t()` will be used in the framework where possible.
And we recommend to use it this way.

It is more "clean" because `Yii::t()` is not actually simple static stateless function, it just hides from you that some global `Application` and `I18N` objects are used behind the scene.
Getting `$this->app` with constructor DI and then calling `$this->app->t()` makes everything explicit hopefully without making troubles. But you are free to choose :)

I'm trying to bring modern approaches and more "clean" OOP while keeping "Yes, it is" style and soul.

P.S.
Finishing Yii 3.0 changes description. It will be available soon.

**Comment by slinstj** [comment]
Thank you @gagatust and @hiqsol. 

I meant in my comment (maybe I had not been clear) was about remove "all" utilities/helpers for the sake of "destatification". This is what I think that would be "over-engineering". But in the answers I see this doesn't seem to be the intention. 👍
